examples/prometheus-agent.jsonnet

```diff
@@ -18,10 +18,22 @@ local kp =
     prometheus+: {
       prometheus+: {
         spec+: {
+          replicas: 1,
           alerting:: {},
           remoteWrite: [{
             url: 'http://remote-write-url.com',
           }],
+          containers+: [
+            {
+              name: 'prometheus',
+              args+: [
+                '--config.file=/etc/prometheus/config_out/prometheus.env.yaml',
+                '--storage.agent.path=/prometheus',
+                '--enable-feature=agent',
+                '--web.enable-lifecycle',
+              ],
+            },
+          ],
```

Comment:
We shouldn't promote overriding those and we should let the operator manage this.

Reply:
Unfortunately, prometheus server and agent use incompatible flags. Is it possible to override the flags in a different way?

Reply:
There is no other way. Seems like we do have a case for faster creation of PrometheusAgent CRD.

I've put this as an agenda item for the next contributor hours. For now, let's can continue with arguments patching to explore what other things need to be adjusted.